### PR TITLE
runfix: split handling quotes and replies

### PR DESCRIPTION
**Pull request by atomrc**
## Description

The process of handling quotes need to be split into 2 different handlers:
- the one that process incoming quotes and decode them
- the one that updates all the messages that are quoting a message that has been updated/deleted

## Checklist

- [x] PR has been self reviewed by the author;
- [x] Hard-to-understand areas of the code have been commented;
- [x] If it is a core feature, unit tests have been added;

### CODE REVIEW

File: src/script/event/preprocessor/QuoteDecoderMiddleware.test.ts [hunk 1]
```diff
@@ -101,57 +100,5 @@ describe('QuotedMessageMiddleware', () => {
...
-      } as any;
-
-      return quotedMessageMiddleware.processEvent(event).then(() => {
-        expect(eventService.replaceEvent).toHaveBeenCalledWith(
-          jasmine.objectContaining({
-            data: jasmine.objectContaining({quote: {error: {type: QuoteEntity.ERROR.MESSAGE_NOT_FOUND}}}),
-          }),
-        );
-      });
-    });
```
Comment (atomrc) [inline src/script/event/preprocessor/QuoteDecoderMiddleware.test.ts:155]: Those features are now covered by the new `RepliesUpdaterMiddleware`

File: src/script/main/app.ts [hunk 2]
```diff
@@ -384,12 +385,14 @@ export class App {
       const serviceMiddleware = new ServiceMiddleware(conversationRepository, userRepository, selfUser);
       const quotedMessageMiddleware = new QuotedMessageMiddleware(this.service.event);
       const readReceiptMiddleware = new ReceiptsMiddleware(this.service.event, conversationRepository, selfUser);
+      const repliesUpdaterMiddleware = new RepliesUpdaterMiddleware(this.service.event);
 
       eventRepository.setEventProcessMiddlewares([
         serviceMiddleware,
         readReceiptMiddleware,
-        eventStorageMiddleware,
         quotedMessageMiddleware,
```
Comment (atomrc) [inline src/script/main/app.ts:393]: decoding quotes need to happen before we store the event in the DB

File: src/script/main/app.ts [hunk 3]
```diff
@@ -384,12 +385,14 @@ export class App {
...
+        eventStorageMiddleware,
+        repliesUpdaterMiddleware,
```
Comment (atomrc) [inline src/script/main/app.ts:395]: But updating the replies to a message needs to happen once the original message is saved in DB

File: src/script/event/preprocessor/QuoteDecoderMiddleware.ts [hunk 4]
```diff
@@ -0,0 +1,103 @@
+/*
+ * Wire
+ * Copyright (C) 2018 Wire Swiss GmbH
```
Comment (V-Gira) [inline src/script/event/preprocessor/QuoteDecoderMiddleware.ts]:
```suggestion
 * Copyright (C) 2023 Wire Swiss GmbH
```
classic

File: src/script/event/preprocessor/QuoteDecoderMiddleware.ts [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+import {getLogger, Logger} from 'Util/Logger';
+import {base64ToArray} from 'Util/util';
+
+import {QuoteEntity} from '../../message/QuoteEntity';
+import {StoredEvent} from '../../storage/record/EventRecord';
+import {ClientEvent} from '../Client';
+import {EventMiddleware, IncomingEvent} from '../EventProcessor';
+import type {EventService} from '../EventService';
+
+export class QuotedMessageMiddleware implements EventMiddleware {
```
Comment (atomrc) [inline src/script/event/preprocessor/QuoteDecoderMiddleware.ts:32]: This entire class is a copy/paste of method from the original `QuotedMessageMIddleware`. Code can be overlooked

File: src/script/event/preprocessor/RepliesUpdaterMiddleware.test.ts [hunk 6]
```diff
@@ -0,0 +1,99 @@
+/*
+ * Wire
+ * Copyright (C) 2018 Wire Swiss GmbH
```
Comment (V-Gira) [inline src/script/event/preprocessor/RepliesUpdaterMiddleware.test.ts]:
```suggestion
 * Copyright (C) 2023 Wire Swiss GmbH
```

**Review by V-Gira — APPROVED** [review]

File: src/script/event/preprocessor/QuoteDecoderMiddleware.test.ts (hunk 1 above)
Comment (V-Gira) [reply src/script/event/preprocessor/QuoteDecoderMiddleware.test.ts:155]: no**w** covered? :D

Comment (atomrc) [reply src/script/event/preprocessor/QuoteDecoderMiddleware.test.ts:155]: oh yes 😅  I wrote exactly the opposite of what I meant 🙃

**Review by PatrykBuniX — APPROVED** [review]

**Pull request merged**